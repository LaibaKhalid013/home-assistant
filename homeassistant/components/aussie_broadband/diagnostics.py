"""Provides diagnostics for Aussie Broadband."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from homeassistant.components.diagnostics import async_redact_data

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant

TO_REDACT = ["address", "ipAddresses", "description", "discounts", "coordinator"]


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    return {
        "services": [
            {
                "service": async_redact_data(service, TO_REDACT),
                "usage": async_redact_data(service["coordinator"].data, ["historical"]),
            }
            for service in hass.data[DOMAIN][config_entry.entry_id]["services"]
        ]
    }
