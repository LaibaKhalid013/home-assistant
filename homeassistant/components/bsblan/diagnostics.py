"""Diagnostics support for BSBLan."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant

    from . import HomeAssistantBSBLANData


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    data: HomeAssistantBSBLANData = hass.data[DOMAIN][entry.entry_id]
    return {
        "info": data.info.to_dict(),
        "device": data.device.to_dict(),
        "state": data.coordinator.data.to_dict(),
    }
