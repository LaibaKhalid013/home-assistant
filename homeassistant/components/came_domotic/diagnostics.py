"""Diagnostics support for Hue."""

from __future__ import annotations

from typing import Any

from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant

from .came_domotic_server import CameDomoticServer
from .const import DOMAIN


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    bridge: CameDomoticServer = hass.data[DOMAIN][entry.entry_id]
    if bridge.api_version == 1:
        # diagnostics is only implemented for V2 bridges.
        return {}
    # Hue diagnostics are already redacted
    return await bridge.api.get_diagnostics()
