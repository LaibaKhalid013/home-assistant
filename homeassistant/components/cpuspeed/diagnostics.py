"""Diagnostics support for CPU Speed."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from cpuinfo import cpuinfo

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    info: dict[str, Any] = cpuinfo.get_cpu_info()
    return info
