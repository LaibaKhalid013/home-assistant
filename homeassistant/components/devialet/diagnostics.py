"""Diagnostics support for Devialet."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from .const import DOMAIN

if TYPE_CHECKING:
    from devialet import DevialetApi

    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    client: DevialetApi = hass.data[DOMAIN][entry.entry_id]

    return await client.async_get_diagnostics()
