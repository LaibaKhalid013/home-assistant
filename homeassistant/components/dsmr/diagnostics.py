"""Diagnostics support for DSMR."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from homeassistant.util.json import json_loads

if TYPE_CHECKING:
    from homeassistant.core import HomeAssistant

    from . import DsmrConfigEntry


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: DsmrConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""

    return {
        "entry": {
            "data": {
                **config_entry.data,
            },
            "unique_id": config_entry.unique_id,
        },
        "data": json_loads(config_entry.runtime_data.telegram.to_json())
        if config_entry.runtime_data.telegram
        else None,
    }
