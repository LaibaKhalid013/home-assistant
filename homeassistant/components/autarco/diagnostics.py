"""Support for the Autarco diagnostics."""

from __future__ import annotations

from typing import Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
from homeassistant.core import HomeAssistant

from . import AutarcoConfigEntry, AutarcoDataUpdateCoordinator

TO_REDACT = {CONF_EMAIL, CONF_PASSWORD}


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: AutarcoConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    autarco_data: list[AutarcoDataUpdateCoordinator] = config_entry.runtime_data

    return {
        "config_entry_data": async_redact_data(dict(config_entry.data), TO_REDACT),
        "sites_data": [
            {
                "id": coordinator.site.site_id,
                "name": coordinator.site.system_name,
                "health": coordinator.site.health,
                "solar": {
                    "power_production": coordinator.data.solar.power_production,
                    "energy_production_today": coordinator.data.solar.energy_production_today,
                    "energy_production_month": coordinator.data.solar.energy_production_month,
                    "energy_production_total": coordinator.data.solar.energy_production_total,
                },
            }
            for coordinator in autarco_data
        ],
    }
