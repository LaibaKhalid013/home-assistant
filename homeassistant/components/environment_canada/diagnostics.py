"""Diagnostics support for Environment Canada."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant

TO_REDACT = {CONF_LATITUDE, CONF_LONGITUDE}


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    coordinators = hass.data[DOMAIN][config_entry.entry_id]
    weather_coord = coordinators["weather_coordinator"]

    return {
        "config_entry_data": async_redact_data(dict(config_entry.data), TO_REDACT),
        "weather_data": dict(weather_coord.ec_data.conditions),
    }
