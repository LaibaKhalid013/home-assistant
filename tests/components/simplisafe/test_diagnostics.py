"""Test SimpliSafe diagnostics."""

from syrupy import SnapshotAssertion

from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant

from tests.components.diagnostics import snapshot_get_diagnostics_for_config_entry
from tests.typing import ClientSessionGenerator


async def test_entry_diagnostics(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    hass_client: ClientSessionGenerator,
    snapshot: SnapshotAssertion,
    setup_simplisafe,
) -> None:
    """Test config entry diagnostics."""
    await snapshot_get_diagnostics_for_config_entry(
        hass, hass_client, config_entry, snapshot
    )
