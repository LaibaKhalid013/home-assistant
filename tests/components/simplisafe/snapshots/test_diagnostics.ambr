# serializer version: 1
# name: test_entry_diagnostics
  dict({
    'entry': dict({
      'data': dict({
        'token': '**REDACTED**',
        'username': '**REDACTED**',
      }),
      'disabled_by': None,
      'domain': 'simplisafe',
      'minor_version': 1,
      'options': dict({
        'code': '**REDACTED**',
      }),
      'pref_disable_new_entities': False,
      'pref_disable_polling': False,
      'source': 'user',
      'title': '**REDACTED**',
      'unique_id': '**REDACTED**',
      'version': 1,
    }),
    'subscription_data': dict({
      '12345': dict({
        'activated': 1445034752,
        'autoActivation': None,
        'billDate': 1602887552,
        'billInterval': 2628000,
        'canceled': 0,
        'country': 'US',
        'creditCard': '**REDACTED**',
        'currency': 'USD',
        'dcid': 0,
        'dispatcher': 'cops',
        'expires': '**REDACTED**',
        'extraTime': 0,
        'features': dict({
          'alerts': True,
          'cameras': 10,
          'careVisit': False,
          'discount': 0,
          'dispatch': True,
          'hazard': True,
          'medical': True,
          'monitoring': True,
          'online': True,
          'proInstall': False,
          'storageDays': 30,
          'video': True,
          'vipCS': False,
        }),
        'location': '**REDACTED**',
        'paymentProfileId': '**REDACTED**',
        'pinUnlocked': True,
        'pinUnlockedBy': 'pin',
        'planName': 'Interactive Monitoring',
        'planSku': 'SSEDSM2',
        'price': 24.99,
        'sStatus': 20,
        'sid': '**REDACTED**',
        'status': dict({
          'hasBaseStation': True,
          'isActive': True,
          'monitoring': 'Active',
        }),
        'subscriptionFeatures': dict({
          'alarmModes': list([
            'Alarm',
            'SecretAlert',
            'Disabled',
          ]),
          'alertChannels': list([
            'Push',
            'SMS',
            'Email',
          ]),
          'alertTypes': list([
            'Alarm',
            'Error',
            'Activity',
            'Camera',
          ]),
          'cameraFeatures': dict({
            'liveView': True,
            'maxRecordingCameras': 10,
            'recordingStorageDays': 30,
            'videoVerification': True,
          }),
          'cellCommunicationBackup': True,
          'dispatchTypes': list([
            'Police',
            'Fire',
            'Medical',
            'Guard',
          ]),
          'monitoredPanicConditions': list([
            'Fire',
            'Medical',
            'Duress',
          ]),
          'monitoredSensorsTypes': list([
            'Entry',
            'Motion',
            'GlassBreak',
            'Smoke',
            'CO',
            'Freeze',
            'Water',
          ]),
          'remoteControl': list([
            'ArmDisarm',
            'LockUnlock',
            'ViewSettings',
            'ConfigureSettings',
          ]),
          'support': dict({
            'annualVisit': False,
            'level': 'Basic',
            'professionalInstall': False,
          }),
          'supportedIntegrations': list([
            'GoogleAssistant',
            'AmazonAlexa',
            'AugustLock',
          ]),
          'timeline': dict({
          }),
        }),
        'time': 2628000,
        'uid': '**REDACTED**',
      }),
    }),
    'systems': list([
      dict({
        'address': '**REDACTED**',
        'alarm_duration': 240,
        'alarm_going_off': False,
        'alarm_volume': 3,
        'battery_backup_power_level': 5293,
        'cameras': list([
          dict({
            'camera_settings': dict({
              'admin': dict({
                'IRLED': 0,
                'audioChannelNum': 2,
                'audioChunkBytes': 2048,
                'audioDetectionEnabled': False,
                'audioDirection': 0,
                'audioSampleFormat': 3,
                'audioSampleRate': 16000,
                'audioSensitivity': 50,
                'audioThreshold': 50,
                'battery': list([
                ]),
                'bitRate': 284,
                'camAgentVersion': '',
                'dbm': 0,
                'enableWifiAnalyticsLib': False,
                'firmwareGroup': 'public',
                'firmwareVersion': '2.6.1.107',
                'fps': 20,
                'gopLength': 40,
                'idr': 1,
                'irCloseDelay': 3,
                'irCloseThreshold': 840,
                'irOpenDelay': 3,
                'irOpenThreshold': 445,
                'irThreshold1x': 388,
                'irThreshold2x': 335,
                'irThreshold3x': 260,
                'ivLicense': '',
                'kframe': 1,
                'lastLogin': 1600639997,
                'lastLogout': 1600639944,
                'levelChangeDelayOne': 10,
                'levelChangeDelayZero': 30,
                'logEnabled': True,
                'logLevel': 3,
                'logQDepth': 20,
                'longPress': 2000,
                'lux': 'lowLux',
                'motionDetectionEnabled': False,
                'motionThresholdOne': 10000,
                'motionThresholdZero': 0,
                'netConfigVersion': '',
                'odAnalyticsLib': 2,
                'odBackgroundLearnStationary': True,
                'odBackgroundLearnStationarySpeed': 15,
                'odCameraFOV': 2,
                'odCameraView': 3,
                'odClassificationConfidenceThreshold': 0.95,
                'odClassificationMask': 2,
                'odClassifierQualityProfile': 1,
                'odEnableObjectDetection': True,
                'odEnableOverlay': False,
                'odEnableVideoAnalyticsWhileStreaming': False,
                'odEventObjectMask': 2,
                'odFGExtractorMode': 2,
                'odLuxHysteresisHigh': 4,
                'odLuxHysteresisLow': 4,
                'odLuxSamplingFrequency': 30,
                'odLuxThreshold': 445,
                'odObjectMinHeightPercent': 24,
                'odObjectMinWidthPercent': 6,
                'odProcessingFps': 8,
                'odSceneType': 1,
                'odSensitivity': 85,
                'odVideoScaleFactor': 1,
                'pirFilterCoefficient': 1,
                'pirHysteresisHigh': 2,
                'pirHysteresisLow': 10,
                'pirSampleRateMs': 800,
                'pirSens': 0,
                'region': 'us-east-1',
                'resSet': 10540,
                'rssi': list([
                  list([
                    1600935204,
                    -43,
                  ]),
                ]),
                'sarlaccDebugLogTypes': 0,
                'statsPeriod': 3600000,
                'statusLEDState': 1,
                'uptime': 810043.74,
                'vmUse': 161592,
                'wifiDisconnects': 1,
                'wifiDriverReloads': 1,
                'wlanMac': 'XX:XX:XX:XX:XX:XX',
              }),
              'cameraName': 'Camera',
              'canRecord': False,
              'canStream': False,
              'doorbellChimeVolume': 'off',
              'enableDoorbellNotification': True,
              'hdr': False,
              'maxDigitalZoom': 10,
              'micEnable': True,
              'micSensitivity': 100,
              'motionSensitivity': 0,
              'nightVision': 'auto',
              'notificationsEnable': False,
              'odLevel': 'medium',
              'pictureQuality': '720p',
              'pirEnable': True,
              'pirLevel': 'medium',
              'privacyEnable': False,
              'shutterAway': 'open',
              'shutterHome': 'closedAlarmOnly',
              'shutterOff': 'closedAlarmOnly',
              'speakerVolume': 75,
              'statusLight': 'off',
              'supportedResolutions': list([
                '480p',
                '720p',
              ]),
              'vaEnable': True,
              'vaZoningCols': 0,
              'vaZoningEnable': False,
              'vaZoningMask': list([
              ]),
              'vaZoningRows': 0,
              'wifiSsid': '',
            }),
            'camera_type': 0,
            'name': 'Camera',
            'serial': '**REDACTED**',
            'shutter_open_when_away': True,
            'shutter_open_when_home': False,
            'shutter_open_when_off': False,
            'status': 'online',
            'subscription_enabled': True,
          }),
        ]),
        'chime_volume': 2,
        'connection_type': 'wifi',
        'entry_delay_away': 30,
        'entry_delay_home': 30,
        'exit_delay_away': 60,
        'exit_delay_home': 0,
        'gsm_strength': -73,
        'light': True,
        'locks': list([
          dict({
            'disabled': False,
            'error': False,
            'lock_low_battery': False,
            'low_battery': False,
            'name': 'Front Door',
            'offline': False,
            'pin_pad_low_battery': False,
            'pin_pad_offline': False,
            'serial': '**REDACTED**',
            'settings': dict({
              'autoLock': 3,
              'away': 1,
              'awayToOff': 0,
              'home': 1,
              'homeToOff': 1,
            }),
            'state': 1,
            'type': 16,
          }),
        ]),
        'notifications': list([
        ]),
        'offline': False,
        'power_outage': False,
        'rf_jamming': False,
        'sensors': list([
          dict({
            'error': False,
            'low_battery': False,
            'name': 'Fire Door',
            'offline': False,
            'serial': '**REDACTED**',
            'settings': dict({
              'away': 1,
              'away2': 1,
              'home': 1,
              'home2': 1,
              'instantTrigger': False,
              'off': 0,
            }),
            'trigger_instantly': False,
            'triggered': False,
            'type': 5,
          }),
          dict({
            'error': False,
            'low_battery': False,
            'name': 'Front Door',
            'offline': False,
            'serial': '**REDACTED**',
            'settings': dict({
              'away': 1,
              'away2': 1,
              'home': 1,
              'home2': 1,
              'instantTrigger': False,
              'off': 0,
            }),
            'trigger_instantly': False,
            'triggered': False,
            'type': 12,
          }),
        ]),
        'serial': '**REDACTED**',
        'state': 99,
        'system_id': '**REDACTED**',
        'temperature': 67,
        'version': 3,
        'voice_prompt_volume': 2,
        'wall_power_level': 5933,
        'wifi_ssid': '**REDACTED**',
        'wifi_strength': -49,
      }),
    ]),
  })
# ---
